source/extensions/filters/common/rbac/engine_impl.cc

```diff
@@ -80,6 +106,64 @@ bool RoleBasedAccessControlEngineImpl::checkPolicyMatch(
   return matched;
 }
 
+RoleBasedAccessControlMatcherEngineImpl::RoleBasedAccessControlMatcherEngineImpl(
+    const xds::type::matcher::v3::Matcher& matcher,
+    Server::Configuration::ServerFactoryContext& factory_context,
+    ActionValidationVisitor& validation_visitor, const EnforcementMode mode)
+    : mode_(mode) {
+  ActionContext context{false};
+  Envoy::Matcher::MatchTreeFactory<Matching::MatchingData, ActionContext> factory(
+      context, factory_context, validation_visitor);
+  matcher_ = factory.create(matcher)();
+  has_log_ = context.has_log_;
+
+  if (!validation_visitor.errors().empty()) {
+    // TODO(snowp): Output all violations.
```

Comment:
nit: TODO

Reply:
I may miss something but it is not a typo?

Reply:
Is this code copied from somewhere? I was suggesting your user name as TODO.

Reply:
Yes, it is a copy. I will fix that.